EuclideanGeometry/Foundation/Axiom/Linear/Ray.lean

```diff
@@ -257,19 +292,70 @@ section existence
 
 -- Archimedean property I : given a directed segment AB (with A ≠ B), then there exists a point P such that B lies on the directed segment AP and P ≠ B.
 
-theorem Seg_nd.exist_pt_beyond_pt {P : Type _} [EuclideanPlane P] (l : Seg_nd P) : (∃ q : P, l.1.target LiesInt (SEG l.1.source q)) := by sorry
-
+theorem Seg_nd.exist_pt_beyond_pt {P : Type _} [EuclideanPlane P] (l : Seg_nd P) : (∃ q : P, l.1.target LiesInt (SEG l.1.source q)) := by 
+  let h := l.1.toVec +ᵥ l.1.target
+  let half : ℝ := 1/2
+  have c: 0 ≤ half ∧ half ≤ 1 ∧ VEC l.1.source l.1.target = half • VEC l.1.source h := by
+    norm_num
+    rw [seg_toVec_eq_vec]
+    repeat rw [Vec.mk_pt_pt]
+    rw [vadd_vsub_assoc]
+    let t := l.1.target -ᵥ l.1.source
+    have x: t = 1/2 * (t + t) := by
+      calc t = 1/2 * (2 * t) := by simp
+      _ = 1/2 * (t + t) := by rw [two_mul]
+    exact x
+  have a: l.1.target LiesOn SEG l.1.source h := ⟨half, c⟩
+  have b: l.1.target ≠ l.1.source ∧ l.1.target ≠ h := by
+    constructor
+    exact l.2
+    have x: l.1.toVec ≠ 0 := by 
+      rw [seg_toVec_eq_vec]
+      rw [Vec.mk_pt_pt]
+      rw [vsub_ne_zero]
+      exact l.2
+    have y: l.1.target ≠ l.1.toVec +ᵥ l.1.target := by
+      rw [ne_comm]
+      by_contra t
+      simp at t
+      exact x t
+    exact y
+  have k: l.1.target LiesInt SEG l.1.source h := ⟨a, b⟩
+  use h
+ 
 -- Archimedean property II: On an nontrivial directed segment, one can always find a point in its interior.  `This will be moved to later disccusion about midpoint of a segment, as the midpoint is a point in the interior of a nontrivial segment`
 
-theorem nd_of_exist_int_pt (l : Seg P) (p : P) (h : p LiesInt l) : l.is_nd := sorry
+theorem nd_of_exist_int_pt (l : Seg P) (p : P) (h : p LiesInt l) : l.is_nd := by
+  rw [Seg.is_nd]
+  rcases h with ⟨⟨c, d⟩, b⟩
+  rcases b with ⟨p_ne_s, _⟩
+  rcases d with ⟨_, _, e⟩
+  have t: VEC Seg.source p ≠ 0 := by exact Iff.mp (ne_iff_vec_ne_zero Seg.source p) p_ne_s
+  have u: c • VEC Seg.source Seg.target ≠ 0 := by 
+    rw [← e]
+    exact t
+  have v: VEC Seg.source Seg.target ≠ 0 := by
+    exact right_ne_zero_of_smul u
+  exact Iff.mp vsub_ne_zero v
```

Comment:
You may write this as `exact Iff.mp vsub_ne_zero (right_ne_zero_of_smul u)` to make prove shorter.